workers/unity/Assets/Playground/Scripts/Player/VirtualJoystick.cs

```diff
@@ -0,0 +1,49 @@
+﻿using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public class VirtualJoystick : MonoBehaviour, IDragHandler, IPointerUpHandler, IPointerDownHandler
+{
+    private Image bgImg;
+    private Image joystickImg;
+
+    public Vector3 InputDirection;
+
+    public void Start()
+    {
+        bgImg = GetComponent<Image>();
+        joystickImg = transform.GetChild(0).GetComponent<Image>();
+        InputDirection = Vector3.zero;
+    }
+
+    public void OnDrag(PointerEventData eventData)
+    {
+        Vector2 pos = Vector2.zero;
+        if (RectTransformUtility.ScreenPointToLocalPointInRectangle(bgImg.rectTransform, eventData.position,
+            eventData.pressEventCamera, out pos))
```

Comment:
Use `out var pos` here ;) C# 7 features, can get rid of the declaration above the if